superflore/test_integration/gentoo/main.py

```diff
@@ -68,9 +93,25 @@ def main():
         for distro in args.ros_distro:
             for pkg in args.pkgs:
                 tester.add_target(distro, pkg)
-    else:
+    elif not args.ci_mode:
         parser.error('Invalid args! You must supply a package list.')
         sys.exit(1)
+    if args.set_upstream:
+        tester.set_upstream(args.set_upstream, args.branch or 'master')
+        repo_org, repo_name = url_to_repo_org(args.set_upstream)
+    if args.ci_mode:
```

Comment:
Is there a reason this work happens down here instead of as part of the [conditional that starts on line 84](https://github.com/ros-infrastructure/superflore/pull/151/files#diff-2939d03ac64407a05976d9d183b47b35R84)?

I stumbled when reading the earlier conditional because it wasn't clear how ci_mode got its test targets. If that conditional read something like

```python
if args.f:
   #args.f things
elif args.ci_mode:
  # args.ci_mode things from this second conditional
else:
  #error out with help message
```

I think that the flow would be more readable.

Reply:
Here I agree that reordering the branches would lead to a higher readability of the code.
```python
    # Environment Setup
    if args.set_upstream:
        ....
    # Target Setup
    if args.f:
        ....
    elif args.pkgs:
        ....
    elif args.ci_mode:
        ....
    # No target given
    else:
        # Help message
        sys.exit(1)

    # Run Tests
    results = tester.run(args.verbose, args.log_file)
```

Further their could be an error state added if an empty yaml file is provided.